pkg/ksync/cli.go

```diff
@@ -11,39 +11,51 @@ import (
 	"github.com/jenkins-x/jx/pkg/util"
 )
 
-// InstallKSync install ksync
-func InstallKSync() (string, error) {
+type KSyncer interface {
```

Comment:
What is the aim of KSyncer? By the name, I would expect that it encapsulate ksync commands, similar to _Kustomizer_ or _Helmer_, but that does not necessarily be the case here.

Reply:
The idea was to break the monolith InstallKSync function and decouple calls to external packages, and make it easier to test (It's quite difficult to write tests in the current codebase), but I think it would make more sense to encapsulate ksync commands in that interface. I will refactor it.

Reply:
> The idea was to break the monolith InstallKSync function and decouple calls to external packages, and make it easier to test 

I am very much for breaking things up. As you say it is all very intervened. I think, however, in the case if the _KSyncInstall_ function you already get a long way with extracting some functions. _KSyncer_ really sends the wrong message, since by name I would expect it to do something differently, especially given the rest of the codebase

> It's quite difficult to write tests in the current codebase

Ohhh yes. I feel the pain. The install command itself should be quite easy though. Look as well at _kustomize_integration_test.go_. Really I think for testing the install command, we do want to download the actual binary. Only this way we can be sure it works. Mocking this is no guarantee that for example, the URL is correct. Let's try to keep the necessary change minimal here. 

> but I think it would make more sense to encapsulate ksync commands in that interface. I will refactor it.

that might be suitable for another issue/pr.

Reply:
I agree abt the integration tests, I will fix this PR this weekend, and get back to you :)

Reply:
The integration tests now check if the binary is installed in jx bin, and the ksync pods are in the kubernetes cluster (this is the area which was the main focus of this PR, the wrong tag for the image resulted in unhealthy pods)